File: 0383-ransom-note/0383-ransom-note.ts
function canConstruct(ransomNote: string, magazine: string): boolean {
    // const passed: {} = {};
    // for (const letter of ransomNote) passed[letter] ? passed[letter]++ : passed[letter] = 1;
    // for (const letter of magazine) passed[letter] ? passed[letter] > 1 ? passed[letter]-- : delete passed[letter] : null;

    // return !Object.keys(passed).length;

    for (const letter of magazine) {
        ransomNote = ransomNote.replace(letter, '');
    };

    return !ransomNote.length;
};

File: 0383-ransom-note/0383-ransom-note.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        dict1 = {}
        for item in magazine:
            if item not in dict1:
                dict1[item] = 1
            else:
                dict1[item] += 1
        for item in ransomNote:
            if item not in dict1:
                return False
            else:
                dict1[item] -= 1

            if dict1[item] < 0:
                return False
        
        return True